src/spi.rs

```diff
@@ -200,6 +204,17 @@ macro_rules! hal {
                 pub fn free(self) -> ($SPIX, (SCK, MISO, MOSI)) {
                     (self.spi, self.pins)
                 }
+
+                pub fn reclock<F>(&mut self, freq: F, clocks: Clocks)
+                    where F: Into<Hertz>
+                {
+                    self.spi.cr1.modify(|_, w| w.spe().disabled());
```

Comment:
Are there any side effects changing the frequency, or is it sufficient to disable and enable it right after it? Any delay needed?

Reply:
Unfortunately I didn't find any documentation about this. 
All I can say is that I tested this with an SD card using the embedded-sdmmc library. I initialize the card at 400KHz, then I switch to 16Mhz and data is red without a problem. I'm reading mp3s from the SD-card and playing them (after decoding) at the same time. Would there be a problem I think I'd notice it. If I leave the card at 400Khz the playing doesn't go well. I tried this also by initializing the card directly at 8 Mhz, and I would get random things and strange crashes in the sdmmc library. I also believe this depends on the device at the other end. With SD cards this works, with other devices it might not.